modules/c++/math.linear/include/math/linear/MatrixMxN.h

```diff
@@ -39,7 +39,17 @@ template<typename _T> bool equals(const _T& e1, const _T& e2)
 
 template<typename _T> inline bool equals(const _T& e1, const _T& e2, _T eps)
 {
+#ifdef _MSC_VER // Visual Studio
```

Comment:
I would rather fix this as described here: http://stackoverflow.com/questions/25573996/c4127-conditional-expression-is-constant; i.e., use template magic so that everything is evaluated at compile-time--which from the warning we already know will work.

But that seemed like too much code for just this one place, and it looks like std::conditional<> in C++14 might provide a better fix: http://en.cppreference.com/w/cpp/types/conditional

If a conditional expression really is constant at compile time, why is it a conditional at all?  Nothing is going to change during program execution, and the condition won't vary.